File: src/Wypozyczenie.java
import java.io.Serializable;
import java.time.LocalDate;

public class Wypozyczenie implements Serializable {
		
	private static final long serialVersionUID = 1L;
	private long id;
	private Ksiazka ksiazka;
	private Czytelnik czytelnik;
	private LocalDate data_wypozyczenia, termin_zwrotu;

	public Wypozyczenie(Ksiazka ksiazka, Czytelnik czytelnik, long id) {
		this.ksiazka = ksiazka;
		this.czytelnik = czytelnik;
		this.id = id;
		this.data_wypozyczenia = LocalDate.now();
		this.termin_zwrotu = LocalDate.now().plusMonths(1);
	}

	public Ksiazka getKsiazka() {
		return ksiazka;
	}

	public Long getKsiazkaID() {
		return ksiazka.getId();
	}

	public void setKsiazka(Ksiazka ksiazka) {
		this.ksiazka = ksiazka;
	}

	public Czytelnik getCzytelnik() {
		return czytelnik;
	}

	public Long getCzytelnikID() {
		return czytelnik.getId();
	}

	public void setCzytelnik(Czytelnik czytelnik) {
		this.czytelnik = czytelnik;
	}

	public long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}

	public LocalDate getDataWypozyczenia() {
		return data_wypozyczenia;
	}

	public LocalDate getTerminZwrotu() {
		return termin_zwrotu;
	}

	public boolean przeterminowane() {
		if (LocalDate.now().isAfter(termin_zwrotu)) {
			return true;
		} else {
			return false;
		}
	}

	@Override
	public String toString() {
		return Long.toString(id) + "	" + ksiazka.getAutor() + "	" + ksiazka.getTytul() + "	" + czytelnik.getImieINazwisko() + "	" + getDataWypozyczenia().toString() + "	" + getTerminZwrotu().toString();
	}
}

File: src/wypozyczenie.cpp
#include "wypozyczenie.h"

Wypozyczenie::Wypozyczenie(quint32 id, quint32 id_ksiazka, quint32 id_czytelnik, QDate data_wypozyczenia) :
    id(id),
    id_ksiazka(id_ksiazka),
    id_czytelnik(id_czytelnik),
    data_wypozyczenia(data_wypozyczenia),
    data_oddania(QDate())
{
}

Wypozyczenie::Wypozyczenie(quint32 id, quint32 id_ksiazka, quint32 id_czytelnik, QDate data_wypozyczenia, QDate data_oddania, bool oddane) :
    oddane(oddane),
    id(id),
    id_ksiazka(id_ksiazka),
    id_czytelnik(id_czytelnik),
    data_wypozyczenia(data_wypozyczenia),
    data_oddania(data_oddania)
{
}

quint32 Wypozyczenie::getID() {
    return id;
}

quint32 Wypozyczenie::getKsiazkaID() {
    return id_ksiazka;
}

quint32 Wypozyczenie::getCzytelnikID() {
    return id_czytelnik;
}

QDate Wypozyczenie::getDataWypozyczenia() {
    return data_wypozyczenia;
}

QDate Wypozyczenie::getDataOddania() {
    return data_oddania;
}

int Wypozyczenie::zwroc() {
    if (!data_oddania.isValid()) {
        data_oddania = QDate::currentDate();
        oddane = true;
        return 0;
    } else {
        return 1;
    }

}
